### PR TITLE
IBX-7418: `ContentName` search criterion added

### PR DESCRIPTION
**Pull request by julitafalcondusza**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | (https://github.com/ibexa/core/pull/312)
| Versions      | master

`ContentName` search criterion added in dev-doc.

#### Checklist

- [x] Text renders correctly
- [x] Text has been checked with vale
- [x] Description metadata is up to date
- [x] Redirects cover removed/moved pages
- [x] Code samples are working
- [ ] PHP code samples have been fixed with PHP CS fixer

### CODE REVIEW

File: docs/search/criteria_reference/contentname_criterion.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+### REST API
+
+=== "XML"
+
+    ```xml
+    <Query>
+        <Filter>
+            <ContentNameCriterion>laptop</ContentNameCriterion>
+        </Filter>
```
Comment (adriendupuis) [inline docs/search/criteria_reference/contentname_criterion.md:26]: I may try to have the REST API examples doing the same than the PHP example.

**Review by ciastektk — CHANGES_REQUESTED** [review]
> I think the example is wrong, where is it coming from?
> 
> According to the constructor only string value is accepted - meaning that multiple values (arrays) are not valid arguments (see: https://github.com/ibexa/core/blob/main/src/contracts/Repository/Values/Content/Query/Criterion/ContentName.php#L16)

I confirm what Marek wrote. 

Why do we have such inconsistency? In description `int(s) representing the Content name`. Next in PHP example array of strings `['laptop', 'tablet']`. Finally, REST examples with a single string value.
Examples from original PR (https://github.com/ibexa/core/pull/312) are clear. String value supporting wildcards.

File: docs/search/criteria_reference/contentname_criterion.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# ContentName Criterion
+
+The [`ContentName` Search Criterion](https://github.com/ibexa/core/blob/main/src/contracts/Repository/Values/Content/Query/Criterion/ContentName.php)
+searches for content by its name.
+
+## Arguments
+
+- `value` - int(s) representing the Content name
```
Comment (adriendupuis) [inline docs/search/criteria_reference/contentname_criterion.md:8]:
```suggestion
- `value` - string(s) representing the Content name(s)
```

File: docs/search/criteria_reference/contentname_criterion.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+### REST API
+
+=== "XML"
+
+    ```xml
+    <Query>
+        <Filter>
+            <ContentNameCriterion>laptop</ContentNameCriterion>
+        </Filter>
+    </Query>
```
Comment (adriendupuis) [inline docs/search/criteria_reference/contentname_criterion.md:27]:
```suggestion
    <Query>
        <Filter>
            <OR>
                <ContentNameCriterion>laptop</ContentNameCriterion>
                <ContentNameCriterion>tablet</ContentNameCriterion>
            </OR>
        </Filter>
    </Query>
```

File: docs/search/criteria_reference/contentname_criterion.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    ```
+
+=== "JSON"
+
+    ```json
+    "Query": {
+        "Filter": {
+            "ContentNameCriterion": "laptop"
+        }
+    }
```
Comment (adriendupuis) [inline docs/search/criteria_reference/contentname_criterion.md:37]:
```suggestion
    "Query": {
        "Filter": {
            "ContentNameCriterion": ["laptop", "tablet"]
        }
    }
```

File: docs/search/criteria_reference/contentname_criterion.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+## Arguments
+
+- `value` - string(s) representing the Content name(s), the joker `*` can be used for partial search.
+
+## Example
+
+### PHP
+
+``` php
+$query->query = new Criterion\ContentName(['laptop', 'tablet']);
```
Comment (alongosz) [inline docs/search/criteria_reference/contentname_criterion.md:15]: Seems the mentioned criterion accepts a single string only: https://github.com/ibexa/core/blob/main/src/contracts/Repository/Values/Content/Query/Criterion/ContentName.php#L16

Comment (adriendupuis) [reply docs/search/criteria_reference/contentname_criterion.md:15]: So, an example illustrting the joker, and that match phone and smart phone.
```suggestion
$query->query = new Criterion\ContentName('*phone');
```

File: docs/search/criteria_reference/contentname_criterion.md [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+=== "XML"
+
+    ```xml
+    <Query>
+        <Filter>
+            <OR>
+                <ContentNameCriterion>laptop</ContentNameCriterion>
+                <ContentNameCriterion>tablet</ContentNameCriterion>
+            </OR>
```
Comment (adriendupuis) [inline docs/search/criteria_reference/contentname_criterion.md:28]:
```suggestion
            <ContentNameCriterion>*phone</ContentNameCriterion>
```

File: docs/search/criteria_reference/contentname_criterion.md [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+        </Filter>
+    </Query>
+    ```
+
+=== "JSON"
+
+    ```json
+    "Query": {
+        "Filter": {
+            "ContentNameCriterion": ["laptop", "tablet"]
```
Comment (adriendupuis) [inline docs/search/criteria_reference/contentname_criterion.md:38]:
```suggestion
            "ContentNameCriterion": "*phone"
```

**Review by mnocon — COMMENTED** [review]
The example is still not correct.

Your example:
``` php
$query->query = new Criterion\ContentName(['laptop', 'tablet']);
```

passes an array with two values (laptop, tablet)

According to the code only a single value is accepted (https://github.com/ibexa/core/blob/main/src/contracts/Repository/Values/Content/Query/Criterion/ContentName.php#L16) (`string $value` means that arrays are not allowed).

Please take a look at the examples in https://github.com/ibexa/core/pull/312 - single arguments are given.

```
new ContentName('phrase'); 

new ContentName('phra*'); 

new ContentName('*phrase*'); 

new ContentName('phr*se*'); 
```
are all valid

File: docs/search/criteria_reference/contentname_criterion.md [hunk 8]
```diff
@@ -0,0 +1,41 @@
+# ContentName Criterion
+
+The [`ContentName` Search Criterion](https://github.com/ibexa/core/blob/main/src/contracts/Repository/Values/Content/Query/Criterion/ContentName.php)
+searches for content by its name.
+
+## Arguments
+
+- `value` - string(s) representing the Content name(s), the joker `*` can be used for partial search.
```
Comment (mnocon) [inline docs/search/criteria_reference/contentname_criterion.md:8]:
```suggestion
- `value` - string(s) representing the Content name(s), the wildcard character `*` can be used for partial search.
```

It's the first time I see the `joker` name - I thought that wildcard is the common term?

See:
https://learn.microsoft.com/en-us/sql/t-sql/language-elements/percent-character-wildcard-character-s-to-match-transact-sql?view=sql-server-ver16
https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-wildcard-query.html
https://solr.apache.org/guide/8_7/the-standard-query-parser.html#wildcard-searches
https://en.wikipedia.org/wiki/Wildcard_character

File: docs/search/criteria_reference/contentname_criterion.md [hunk 9]
```diff
@@ -0,0 +1,38 @@
+# ContentName Criterion
+
+The [`ContentName` Search Criterion](https://github.com/ibexa/core/blob/main/src/contracts/Repository/Values/Content/Query/Criterion/ContentName.php)
+searches for content by its name.
+
+## Arguments
+
+- `value` - string(s) representing the Content name(s), the wildcard character `*` can be used for partial search.
```
Comment (adriendupuis) [inline docs/search/criteria_reference/contentname_criterion.md:8]:
```suggestion
- `value` - string representing the Content name, the wildcard character `*` can be used for partial search.
```

**Review by adriendupuis — APPROVED** [review]

File: docs/search/criteria_reference/contentname_criterion.md [hunk 10]
```diff
@@ -0,0 +1,38 @@
+# ContentName Criterion
+
+The [`ContentName` Search Criterion](https://github.com/ibexa/core/blob/main/src/contracts/Repository/Values/Content/Query/Criterion/ContentName.php)
+searches for content by its name.
+
+## Arguments
+
+- `value` - string representing the content name, the wildcard character `*` can be used for partial search.
```
Comment (dabrt) [inline docs/search/criteria_reference/contentname_criterion.md:8]:
```suggestion
- `value` - string representing the content name, the wildcard character `*` can be used for partial search
```

**Review by dabrt — APPROVED** [review]

**Review by ciastektk — APPROVED** [review]

**Pull request merged**